### DISCUSSION TITLE
Set bigger margins

### DISCUSSION THREAD

**Discussion by suaviloquence** [koreader/koreader#11100] [Q&A] [answered]
Hi,
Unfortunately, I recently broke the corner of my ereader's screen.  The rest of the screen still functions, however, and I'd like to still be able to read on it, even if the experience is diminished.  I tried to set a left margin so that all the text would be in the visible portion of the screen, but the largest koreader will go is 140, even if I manually edit the docsettings file for the individual book.  I understand that this is not the intended use case, but does anyone have any ideas of how I can increase the margins or otherwise have koreader render the page only in the visible area?
![81D1893D-0C03-41F8-87CA-99D1FE742CBA](https://github.com/koreader/koreader/assets/56054085/202c5eef-ad7e-4015-9a8b-94b86a5bf92d)

**Comment by Frenzie** [comment, marked as answer]
You can't really deal with a square cutout, but by adding a viewport to your relevant device definition you could render exclusively in the lower part of the screen for example.
https://github.com/koreader/koreader/blob/da8e23c0112b87cc75951525905d6094bef0de5c/frontend/device/kobo/device.lua#L215

**Reply by NiLuJe** [reply]
Or on the right side of the screen, pick your poison ;).

(The bottom part, if you then switch to using the device rotated, might be the least terrible approach, though).

**Reply by suaviloquence** [reply]
Thanks both of you for your help! If anyone comes across this in the future, I found my device file (for me, it was `frontend/device/pocketbook/device.lua`), found my device, and added this line to the definition: 
```lua
local PocketBook632 = PocketBook:extend{
    model = "PBTouchHDPlus",
    display_dpi = 300,
    isAlwaysPortrait = yes,
    usingForcedRotation = landscape_ccw,
    hasNaturalLight = yes,
   viewport = Geom:new{x=500,y=0,w=1072-500,h=1448}, -- ++++++++++
}
```
where 500 was the horizontal offset from the left that I wanted, and my device's dimensions are 1072 by 1448.
I also had to add `local Geom = require("ui/geometry")` at the top of the file, since it wasn't imported already for the pocketbook file.
